unit/cq_test.c

```diff
@@ -66,12 +67,49 @@ create_cq(struct fid_cq **cq, size_t size, uint64_t flags,
 	return fi_cq_open(domain, &cq_attr, cq, NULL);
 }
 
+/* Try opening fi->domain_attr->cq_cnt number of completion queues
+ * simultaneously using the maximum size found in the first test.
+ */
+static int cq_open_close_simultaneous(void)
+{
+	int ret;
+	int opened;
+	size_t count;
+	int testret = FAIL;
+	struct fid_cq **cq_array;
+
+	count = fi->domain_attr->cq_cnt;
+	FT_DEBUG("testing creation of up to %zu simultaneous CQs\n", count);
+
+	cq_array = calloc(count, sizeof(*cq_array));
+	if (!cq_array)
+		return -FI_ENOMEM;
+
+	for (opened = 0; opened < count; opened++) {
+		ret = create_cq(&cq_array[opened], maximum_size, 0,
```

Comment:
cq_cnt is defined as the optimal number of CQs that can be opened relative to the default CQ attributes.  This is attempting to open each CQ as the maximum size it can be.  I would replace maximum_size with 0, so that the default can be used.